code/modules/projectiles/magazines/shotguns.dm

```diff
@@ -57,6 +57,102 @@ one type of shotgun ammo, but I think it helps in referencing it. ~N
 	max_rounds = 20 // Real rifle boxes are usually 20 rounds. This works with the new system, four handfuls.
 	w_class = WEIGHT_CLASS_SMALL // CAN throw it in your pocket, friend.
 
+/obj/item/ammo_magazine/rifle/martini
+	name = "box of .557/440 rifle rounds"
+	desc = "A box filled with rifle bullets."
+	icon_state = ".557" 
+	default_ammo = /datum/ammo/bullet/sniper/martini
+	caliber = ".557/440" 
+	gun_type = /obj/item/weapon/gun/shotgun/double/martini
+	max_rounds = 20
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/ammo_magazine/rifle/martini/belt
+	name = "martini henry ammo belt"
+	desc = "A belt for holding all your .577/400 Ball Ammo"
+	icon_state = ".557_belt"
+	default_ammo = /datum/ammo/bullet/sniper/martini
+	caliber = ".557/440"
+	gun_type = /obj/item/weapon/gun/shotgun/double/martini
+	max_rounds = 6
+	flags_equip_slot = ITEM_SLOT_BELT
+
+/obj/item/ammo_magazine/rifle/martini/belt/Initialize(mapload, spawn_empty)
+	. = ..()
+	update_icon()
+
+/obj/item/ammo_magazine/rifle/martini/belt/update_icon()
+	if(current_rounds <= 0)
+		icon_state = initial(icon_state) + "_e"
+		return
+	icon_state = initial(icon_state)
+	setDir(current_rounds + round(current_rounds/3))
+
+///Draws ammo from a magazine into the mob/user 's hand
+/obj/item/ammo_magazine/rifle/martini/belt/proc/draw_ammo(mob/user)
+	var/mob/living/carbon/human/U = user
+	if(src != U.belt)
+		return FALSE
+	if (current_rounds > 0)
+		if(create_handful(U, 1))
+			return TRUE
+	to_chat(U, "[src] is empty. Nothing to grab.")
+	return TRUE
+			
+///Overwrites regular magazine draw proc to instead draw a single bullet and allow it to be worn and operated
+/obj/item/ammo_magazine/rifle/martini/belt/attack_hand(mob/living/user)
+	if(draw_ammo(user))
+		return
+	..()
+
+///Overwrites regular magazine refill to allow it to be worn and operated
+/obj/item/ammo_magazine/rifle/martini/belt/attackby(obj/item/I, mob/user, params)
+	
+	if(!istype(I, /obj/item/ammo_magazine))
+		return ..()
+	var/mob/living/carbon/human/U = user
+	var/obj/item/ammo_magazine/H = I
```

Comment:
Use better var names for these please, human = U and Ammo magazine = H is difficult to read